src/components/arm/Arm.ts

```diff
@@ -0,0 +1,54 @@
+import type { Pose, WorldState } from '../../gen/common/v1/common_pb.esm';
+
+import type { JointPositions } from '../../gen/component/arm/v1/arm_pb.esm';
+
+import type { Extra } from '../../types';
+
+export interface Arm {
+  /**
+   * Get the position of the end of the arm expressed as a pose
+   *
+   */
+  GetEndPosition: (extra?: Extra) => Promise<Pose>;
+
+  /**
+   * Move the end of the arm to the pose, avoiding obstacles in the worldstate
+   *
+   * @param pose - The destination pose for the arm
+   * @param world - The obstacles for the arm to avoid
+   */
+  MoveToPosition: (
+    pose: Pose,
+    world: WorldState,
+    extra?: Extra
+  ) => Promise<void>;
+
+  /**
+   * Move each joint of the arm based on the angles on the joint poisitons
+   * parameter
+   *
+   * @param jointPositions Destination joint positons
```

Comment:
```suggestion
   * @param jointPositions - Destination joint positons
```

the hyphen is required to display correctly apparently: https://tsdoc.org/pages/tags/param/